community-contributions/telegram-faucet/bot_tg_gettoque/bot.cs

```diff
@@ -0,0 +1,338 @@
+
+            }
+
+
+
+
+        }
+
+        static void Main(string[] args)
+        {
+
+
+            Bot = new TelegramBotClient(""); //api
+
+            Bot.OnMessage += BotOnMessageReceived; //subscribe on receiving msg
+
+            Bot.StartReceiving(); // listening msg
+            var me = Bot.GetMeAsync().Result;
+            Console.ReadLine();
+            Bot.StopReceiving();
+        }
+
+        private static async void BotOnMessageReceived(object sender, Telegram.Bot.Args.MessageEventArgs e)
+        {
+
+            var message = e.Message;
+            Console.WriteLine($"{message.Chat.Id} send {message.Text}");
+            if (message == null || message.Type != MessageType.Text) //check if not nul and text
+            {
+                return;
+            }
+
+            var MainKeboard = new ReplyKeyboardMarkup(
+                   new KeyboardButton[][]
+                   {
+                        new KeyboardButton[] { "💰 Get 101 JOY" }
+
+                   },
+                   resizeKeyboard: true
+               );
+
+            var HelpKeyboard = new ReplyKeyboardMarkup(
+                 new KeyboardButton[][]
+                 {
+                        new KeyboardButton[] { "🆘 Need help?" }
+
+                 },
+                 resizeKeyboard: true
+             );
+
+         
+            Console.WriteLine($"{DateTime.Now:yyyy-MM-dd:hh-mm-ss} {message.Chat.Id} send msg = '{message.Text}'");
+
+            try
+            {
+
+                if (message == null || message.Type != MessageType.Text) return;
+
+                switch (message.Text)
+                {
+                    case "/start":
+                        {
+
+                            //check if number is exist and if start then change to false
+
+
+
+                            await Bot.SendTextMessageAsync(message.Chat.Id, WelcomAndLog(message), replyMarkup: MainKeboard);
+                            break;
+
+                        }
+
+                    case "💰 Get 101 JOY":
+                        {
+
+                            //check if number is exist and if start then change to false
+
+
+                            await Bot.SendTextMessageAsync(message.Chat.Id, "Send your wallet address", replyMarkup:HelpKeyboard);
+                            break;
+
+                        }
+
+
+                    case "🆘 Need help?":
+                        {
+
+                            //check if number is exist and if start then change to false
+
+
+                            await Bot.SendTextMessageAsync(message.Chat.Id, Helpmsg(), replyMarkup: HelpKeyboard);
+                            break;
+
+                        }
+
+
+
+
+
+
+                }
+
+                /*
+                if (message.ReplyToMessage != null && message.ReplyToMessage.Text.Contains("Send your wallet address"))
+                {
+                    GetAddress(message);
+                }
+                */
+
```

Comment:
Commented out functionality doesn't really add anything to the code. I think it would be a good idea to delete these wherever possible if you don't plan on using this functionality in the future.